File: DSA/Pranvkumar_590011587/Day9_Question3.py
from collections import OrderedDict

def first_non_repeating_char(string):
    freq = OrderedDict()
    for consonants in string:
        freq[consonants] = freq.get(consonants, 0) + 1
    for consonants in freq:
        if freq[consonants] == 1:
            return consonants
    return -1


# 3. First non-repeating character
string = input("Enter a string to find the first non-repeating character: ")
result = first_non_repeating_char(string)
print(f'First non-repeating character in "{string}": {result}')

File: DSA/Stuti_Jain_590012149/Day_9_Question_3.c
#include<stdio.h>
char s[100];
int i, len=0, j;
int main() {
    printf("Enter the string: \n");
    scanf("%s", s);

   for(i=0; s[i]; i++){
    len++;
   }

   for(i=0; i<len; i++){
    for(j=0; j<len; j++){
        if(i!=j && s[i]==s[j]){
            break;
        }
    }

    if(j==len){
        printf("the first non-repeating character is: %c", s[i]);

        return 0;
    }
}

printf("-1\n");

return 0;
}

